csaf_2.0/json_schema/csaf_json_schema.json

```
@@ -154,13 +154,127 @@
+                  "algorithm": {
+                    "title": "Algorithm of the cryptographic hash",
+                    "description": "Contains the name of the cryptographic hash algorithm used to calculate the value.",
+                    "type": "string",
+                    "enum": [
+                      "SHA-256",
+                      "SHA-384",
+                      "SHA-512",
+                      "SHA-3",
+                      "BLAKE3"
+                    ]
+                  },
+                  "file": {
+                    "title": "Filename",
+                    "description": "Contains the filename of the file which is identified by the hash value.",
+                    "type": "string",
+                    "minLength": 1,
+                    "examples": [
+                      "WINWORD.EXE",
+                      "msotadddin.dll",
+                      "sudoers.so"
+                    ]
+                  },
+                  "value": {
+                    "title": "Value of the cryptographic hash",
+                    "description": "Contains the cryptographic hash value.",
+                    "type": "string",
+                    "minLength": 64,
+                    "pattern": "^([0-9a-fA-F]){64,}$",
+                    "examples": [
+                      "4775203615d9534a8bfca96a93dc8b461a489f69124a130d786b42204f3341cc",
+                      "9ea4c8200113d49d26505da0e02e2f49055dc078d1ad7a419b32e291c7afebbb84badfbd46dec42883bea0b2a1fa697c",
+                      "37df33cb7464da5c7f077f4d56a32bc84987ec1d85b234537c1c1a4d4fc8d09dc29e2e762cb5203677bf849a2855a0283710f1f5fe1d6ce8d5ac85c645d0fcb3"
+                    ]
+                  }
+                }
+              }
+
+            },
+            "purl": {
+              "$comment": "TODO: This should have a full regular expression to enforce purl syntax.",
+              "title": "package URL representation",
+              "description": "The package URL (purl) attribute refers to a method for reliably identifying and locating software packages external to this specification.",
+              "type": "string",
+              "pattern": "^pkg:",
+              "minLength": 4
+            },
+            "serial_numbers": {
+              "$comment": "TODO: Add remark in prose that this should only be used if a certain range of serial numbers with a software version is affected or the serial numbers change during update.",
+              "title": "List of serial numbers",
+              "description": "Contains a list of parts or a full serial numbers.",
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "title": "Serial number",
+                "description": "Contains a part or a full serial number of the component to identify.",
+                "type": "string",
+                "minLength": 1
+              }
+            },
+            "x_other_ids": {
```

How about:

```suggestion
            "x_generic_uris": {
```

"Other" seems a bit vague, but maybe that's just personal preferences :-)

In the properties, it could also then be `namespace` and `uri`.

---

I like the generic part and dislike other (which has that smell of not caring, whilst we are opening up for extension in the future)

---

If it is a `uri` then the format should also be `uri`. Correct?

---

> I like the generic part and dislike other (which has that smell of not caring, whilst we are opening up for extension in the future)

I agree. Does it actually always have to be an uri?

---

I would go for a URI and hope there will always sufficient automatic resolution of those across the community.
That would allow (citing from RFC 3986 section 1.1.2. Examples:
>ftp://ftp.is.co.za/rfc/rfc1808.txt
>http://www.ietf.org/rfc/rfc2396.txt
>ldap://[2001:db8::7]/c=GB?objectClass?one
>mailto:John.Doe@example.com
>news:comp.infosystems.www.servers.unix
>tel:+1-816-555-1212
>telnet://192.0.2.16:80/
>urn:oasis:names:specification:docbook:dtd:xml:4.1.2

especially URNs often did not help XML editing tools to resolve automatically in practice (on my little planet) - but nevertheless the producer of the CSAF using this field could even say: *Call jane!* (per the `tel:` scheme) :wink: